src/main/scala/firrtl/passes/memlib/DumpMemoryAnnotations.scala

```diff
@@ -13,16 +15,149 @@ class DumpMemoryAnnotations extends Transform with DependencyAPIMigration {
+      s"      ram[initvar] = {${(width - 1) / 32 + 1} {$$random}};"
+    ) ++ readPorts.map(port => s"    reg_${port.prefix}addr = {${(addrWidth - 1) / 32 + 1} {$$random}};") ++ Seq(
+      "  end",
+      "`endif"
+    )
+    val decl = genDecl()
+
+    def genSequential(port: Port): Seq[String] = {
+      def genReadSequential(en: String) = Seq(
+        s"always @(posedge ${port.prefix}clk)",
+        s"  reg_${port.prefix}ren <= $en;",
+        s"always @(posedge ${port.prefix}clk)",
+        s"  if ($en) reg_${port.prefix}addr <= ${port.prefix}addr;"
+      )
+      def genWriteSequential(en: String, inputData: String): Seq[String] = Seq(
+        s"always @(posedge ${port.prefix}clk)",
+        s"  if ($en) begin"
+      ) ++ (0 until maskSeg).map { i =>
+        val mask = if (masked) s"if (${port.prefix}mask[$i]) " else ""
+        val ram_range = s"${(i + 1) * maskGran - 1}:${i * maskGran}"
+        s"    ${mask}ram[${port.prefix}addr][$ram_range] <= ${port.prefix}$inputData[$ram_range];"
+      } ++ Seq("  end")
+
+      port.portType match {
+        case ReadPort  => genReadSequential(port.prefix + "en")
+        case WritePort => genWriteSequential(port.prefix + "en", "data")
+        case ReadWritePort =>
+          genReadSequential(s"${port.prefix}en && !${port.prefix}wmode") ++
+            genWriteSequential(s"${port.prefix}en && ${port.prefix}wmode", "wdata")
+      }
+    }
+    val sequential = ports.flatMap(genSequential)
+
+    def genCombinational(port: Port): Seq[String] = {
+      val data = port.prefix + (if (port.portType == ReadWritePort) "rdata" else "data")
+      Seq(
+        "`ifdef RANDOMIZE_GARBAGE_ASSIGN",
+        s"reg [${((width - 1) / 32 + 1) * 32 - 1}:0] ${port.prefix}random;",
+        "`ifdef RANDOMIZE_MEM_INIT",
+        "  initial begin",
+        "    #`RANDOMIZE_DELAY begin end",
+        s"    ${port.prefix}random = {${Seq.fill((width - 1) / 32 + 1)("$random").mkString(", ")}};",
+        s"    reg_${port.prefix}ren = ${port.prefix}random[0];",
+        "  end",
+        "`endif",
+        s"always @(posedge ${port.prefix}clk) ${port.prefix}random <= {${Seq.fill((width - 1) / 32 + 1)("$random").mkString(", ")}};",
+        s"assign $data = reg_${port.prefix}ren ? ram[reg_${port.prefix}addr] : ${port.prefix}random[${width - 1}:0];",
+        "`else",
+        s"assign $data = ram[reg_${port.prefix}addr];",
+        "`endif"
+      )
+    }
+    val combinational = readPorts.flatMap(genCombinational)
+
+    val body = if (genBlackBox) "" else s"""
+                                           |  ${decl.mkString("\n  ")}
+                                           |  ${sequential.mkString("\n  ")}
+                                           |  ${combinational.mkString("\n  ")}""".stripMargin
+
+    (name, s"""
+              |module $name(
+              |  ${portSpec.mkString(",\n  ")}
+              |);
+              |
+              |$body
+              |
+              |endmodule""".stripMargin)
```

Comment:
Add Verilog comments in the header, since original `config` is not elaborated, need contain `DefAnnotatedMemory` informations

Reply:
Fixed in e0859e8 (#2202)